File: mobile/lib/domain/models/pin_response.dart
/// doc/API_SPEC.md 6 — PinResponse
class PinResponse {
  const PinResponse({
    required this.id,
    required this.latitude,
    required this.longitude,
    required this.ownerId,
    this.ownerNickname,
    this.description,
    this.createdAt,
    this.updatedAt,
  });

  final int id;
  final double latitude;
  final double longitude;
  final int ownerId;
  final String? ownerNickname;
  final String? description;
  final String? createdAt;
  final String? updatedAt;

  /// 하위 호환용 별칭 (과거 필드명: memberId)
  int get memberId => ownerId;

  factory PinResponse.fromJson(Map<String, dynamic> json) {
    return PinResponse(
      id: json['id'] as int,
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      ownerId: (json['ownerId'] ?? json['memberId']) as int,
      ownerNickname: json['ownerNickname'] as String?,
      description: json['description'] as String?,
      createdAt: json['createdAt'] as String?,
      updatedAt: json['updatedAt'] as String?,
    );
  }
}

File: src/main/java/com/example/sns/dto/response/PinResponse.java
package com.example.sns.dto.response;

import java.time.LocalDateTime;

import com.example.sns.domain.Pin;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Pin 응답 DTO.
 * 프론트엔드 호환: userId, userName, lat, lng 별칭 제공.
 */
public record PinResponse(
        Long id,
        Long ownerId,
        String ownerNickname,
        String title,
        String description,
        String category,
        Double latitude,
        Double longitude,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {

    @JsonProperty("userId")
    public Long userId() {
        return ownerId;
    }

    @JsonProperty("userName")
    public String userName() {
        return ownerNickname;
    }

    @JsonProperty("lat")
    public Double lat() {
        return latitude;
    }

    @JsonProperty("lng")
    public Double lng() {
        return longitude;
    }

    public static PinResponse from(Pin pin) {
        return new PinResponse(
                pin.getId(),
                pin.getOwner().getId(),
                pin.getOwner().getNickname(),
                pin.getTitle(),
                pin.getDescription(),
                pin.getCategory(),
                pin.getLatitude(),
                pin.getLongitude(),
                pin.getCreatedAt(),
                pin.getUpdatedAt()
        );
    }
}
